service/frontend/workflow_handler.go

```diff
@@ -5217,6 +5217,9 @@ func (wh *WorkflowHandler) validateWorkflowCompletionCallbacks(
 	}
 
 	for _, callback := range callbacks {
+		if err := wh.validateLinks(ns, callback.GetLinks()); err != nil {
```

Comment:
Now that we support links in both callbacks and outside of them, the validation should limit the total across the two places.